
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Basic example &#8212; dmslogo 0.6.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Plotting heatmap overlays" href="overlays.html" />
    <link rel="prev" title="Examples" href="examples.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container,
div.nbinput.container div.prompt,
div.nbinput.container div.input_area,
div.nbinput.container div[class*=highlight],
div.nbinput.container div[class*=highlight] pre,
div.nboutput.container,
div.nboutput.container div.prompt,
div.nboutput.container div.output_area,
div.nboutput.container div[class*=highlight],
div.nboutput.container div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    width: 4.5ex;
    padding-top: 5px;
    position: relative;
    user-select: none;
}

div.nbinput.container div.prompt > div,
div.nboutput.container div.prompt > div {
    position: absolute;
    right: 0;
    margin-right: 0.3ex;
}

@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        width: unset;
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }

    div.nbinput.container div.prompt > div,
    div.nboutput.container div.prompt > div {
        position: unset;
    }
}

/* disable scrollbars on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }


div.nbinput.container div.input_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight].math,
div.nboutput.container div.output_area.rendered_html,
div.nboutput.container div.output_area > div.output_javascript,
div.nboutput.container div.output_area:not(.rendered_html) > img{
    padding: 5px;
}

/* fix copybtn overflow problem in chromium (needed for 'sphinx_copybutton') */
div.nbinput.container div.input_area > div[class^='highlight'],
div.nboutput.container div.output_area > div[class^='highlight']{
    overflow-y: hidden;
}

/* hide copybtn icon on prompts (needed for 'sphinx_copybutton') */
.prompt a.copybtn {
    display: none;
}

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}
</style>
<div class="section" id="Basic-example">
<h1>Basic example<a class="headerlink" href="#Basic-example" title="Permalink to this headline">¶</a></h1>
<div class="section" id="Import-packages-/-modules">
<h2>Import packages / modules<a class="headerlink" href="#Import-packages-/-modules" title="Permalink to this headline">¶</a></h2>
<p>Import <code class="docutils literal notranslate"><span class="pre">dmslogo</span></code> along with the other Python packages used in these examples:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># NBVAL_IGNORE_OUTPUT</span>

<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">random</span>

<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span><span class="p">,</span> <span class="n">Image</span>

<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="kn">import</span> <span class="nn">numpy</span>

<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="kn">import</span> <span class="nn">dmslogo</span>
<span class="kn">from</span> <span class="nn">dmslogo.colorschemes</span> <span class="kn">import</span> <span class="n">CBPALETTE</span>
</pre></div>
</div>
</div>
<p>Set options to display pandas DataFrames:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">pd</span><span class="o">.</span><span class="n">set_option</span><span class="p">(</span><span class="s1">&#39;display.max_columns&#39;</span><span class="p">,</span> <span class="mi">20</span><span class="p">)</span>
<span class="n">pd</span><span class="o">.</span><span class="n">set_option</span><span class="p">(</span><span class="s1">&#39;display.width&#39;</span><span class="p">,</span> <span class="mi">500</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Simple-example-on-toy-data">
<h2>Simple example on toy data<a class="headerlink" href="#Simple-example-on-toy-data" title="Permalink to this headline">¶</a></h2>
<div class="section" id="Draw-a-basic-logo">
<h3>Draw a basic logo<a class="headerlink" href="#Draw-a-basic-logo" title="Permalink to this headline">¶</a></h3>
<p>Simple plotting can be done using the <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.logo.html#dmslogo.logo.draw_logo">dmslogo.logo.draw_logo</a> function.</p>
<p>This function takes in as input a pandas DataFrame that has columns with:</p>
<ul class="simple">
<li><p>site in sequential integer numbering</p></li>
<li><p>letter (i.e., amino acid or nucleotide)</p></li>
<li><p>height of letter (can be any positive number)</p></li>
</ul>
<p>Here make a simple data frame that fits these specs:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="o">.</span><span class="n">from_records</span><span class="p">(</span>
        <span class="n">data</span><span class="o">=</span>   <span class="p">[(</span>    <span class="mi">1</span><span class="p">,</span>      <span class="s1">&#39;A&#39;</span><span class="p">,</span>        <span class="mi">1</span><span class="p">),</span>
                 <span class="p">(</span>    <span class="mi">1</span><span class="p">,</span>      <span class="s1">&#39;C&#39;</span><span class="p">,</span>      <span class="mf">0.7</span><span class="p">),</span>
                 <span class="p">(</span>    <span class="mi">2</span><span class="p">,</span>      <span class="s1">&#39;C&#39;</span><span class="p">,</span>      <span class="mf">0.1</span><span class="p">),</span>
                 <span class="p">(</span>    <span class="mi">2</span><span class="p">,</span>      <span class="s1">&#39;D&#39;</span><span class="p">,</span>      <span class="mf">1.2</span><span class="p">),</span>
                 <span class="p">(</span>    <span class="mi">5</span><span class="p">,</span>      <span class="s1">&#39;A&#39;</span><span class="p">,</span>      <span class="mf">0.4</span><span class="p">),</span>
                 <span class="p">(</span>    <span class="mi">5</span><span class="p">,</span>      <span class="s1">&#39;K&#39;</span><span class="p">,</span>      <span class="mf">0.4</span><span class="p">)],</span>
        <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;site&#39;</span><span class="p">,</span> <span class="s1">&#39;letter&#39;</span><span class="p">,</span> <span class="s1">&#39;height&#39;</span><span class="p">])</span>

<span class="n">data</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>site</th>
      <th>letter</th>
      <th>height</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>A</td>
      <td>1.0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>C</td>
      <td>0.7</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>C</td>
      <td>0.1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2</td>
      <td>D</td>
      <td>1.2</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>A</td>
      <td>0.4</td>
    </tr>
    <tr>
      <th>5</th>
      <td>5</td>
      <td>K</td>
      <td>0.4</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Use <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.logo.html#dmslogo.logo.draw_logo">dmslogo.logo.draw_logo</a> to draw the logo plot, passing the names of the columns with each piece of required data:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_7_0.png" src="_images/basic_example_7_0.png" />
</div>
</div>
<p>Add a title:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                            <span class="n">title</span><span class="o">=</span><span class="s1">&#39;basic example&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_9_0.png" src="_images/basic_example_9_0.png" />
</div>
</div>
<p>Note that the call to <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.logo.html#dmslogo.logo.draw_logo">dmslogo.logo.draw_logo</a> returns matplotlib Figure and Axis instances, which we have called <code class="docutils literal notranslate"><span class="pre">fig</span></code> and <code class="docutils literal notranslate"><span class="pre">ax</span></code>. We can save the figure to a file using the <a class="reference external" href="https://matplotlib.org/api/_as_gen/matplotlib.pyplot.savefig.html">savefig</a> command of <code class="docutils literal notranslate"><span class="pre">fig</span></code>. Below we show an example of how to do this saving to a temporary file:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">outputdir</span> <span class="o">=</span> <span class="s1">&#39;output_files&#39;</span>
<span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">outputdir</span><span class="p">,</span> <span class="n">exist_ok</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">pngfile</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">outputdir</span><span class="p">,</span> <span class="s1">&#39;basic_example_logo.png&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Saving figure to </span><span class="si">{</span><span class="n">pngfile</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="n">pngfile</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">450</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Here is a rendering of the saved figure:&quot;</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">Image</span><span class="p">(</span><span class="n">pngfile</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">200</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Saving figure to output_files/basic_example_logo.png
Here is a rendering of the saved figure:
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_11_1.png" class="no-scaled-link" src="_images/basic_example_11_1.png" style="width: 200px;" />
</div>
</div>
</div>
<div class="section" id="Drawing-without-breaks">
<h3>Drawing without breaks<a class="headerlink" href="#Drawing-without-breaks" title="Permalink to this headline">¶</a></h3>
<p>Note how the above plot has a “break” (gap and dashed line) to indicate a break in the sequential numbering in <code class="docutils literal notranslate"><span class="pre">x_col</span></code> between 2 and 5. This is useful as it indicates when we are breaking the sequence when drawing just snippets of a protein. If you do <strong>not</strong> want to indicate breaks in this way, turn off the <code class="docutils literal notranslate"><span class="pre">addbreaks</span></code> option. Now the logo just goes directly from 2 to 5 without indicating a break:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                            <span class="n">addbreaks</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_13_0.png" src="_images/basic_example_13_0.png" />
</div>
</div>
</div>
<div class="section" id="Setting-letter-colors">
<h3>Setting letter colors<a class="headerlink" href="#Setting-letter-colors" title="Permalink to this headline">¶</a></h3>
<p>The above plot colored letters using a default amino-acid coloring scheme. You can set a different coloring scheme using <code class="docutils literal notranslate"><span class="pre">colorscheme</span></code> and <code class="docutils literal notranslate"><span class="pre">missing_color</span></code>, or you can set letter colors at a site-specific level by adding a column to <code class="docutils literal notranslate"><span class="pre">data</span></code> that specifies the colors. Here we color letters at site-specific level:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;color&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;red&#39;</span><span class="p">,</span> <span class="s1">&#39;gray&#39;</span><span class="p">,</span> <span class="s1">&#39;gray&#39;</span><span class="p">,</span> <span class="s1">&#39;gray&#39;</span><span class="p">,</span> <span class="s1">&#39;red&#39;</span><span class="p">,</span> <span class="s1">&#39;gray&#39;</span><span class="p">]</span>
<span class="n">data</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>site</th>
      <th>letter</th>
      <th>height</th>
      <th>color</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>A</td>
      <td>1.0</td>
      <td>red</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>C</td>
      <td>0.7</td>
      <td>gray</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>C</td>
      <td>0.1</td>
      <td>gray</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2</td>
      <td>D</td>
      <td>1.2</td>
      <td>gray</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>A</td>
      <td>0.4</td>
      <td>red</td>
    </tr>
    <tr>
      <th>5</th>
      <td>5</td>
      <td>K</td>
      <td>0.4</td>
      <td>gray</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Now plot using <code class="docutils literal notranslate"><span class="pre">color_col</span></code> to set the colors:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                            <span class="n">color_col</span><span class="o">=</span><span class="s1">&#39;color&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_17_0.png" src="_images/basic_example_17_0.png" />
</div>
</div>
</div>
<div class="section" id="Labeling-x-axis-ticks">
<h3>Labeling x-axis ticks<a class="headerlink" href="#Labeling-x-axis-ticks" title="Permalink to this headline">¶</a></h3>
<p>Sometimes we want to label sites with something other than the sequential integer numbers. We can do this by adding a column for the xtick labels to <code class="docutils literal notranslate"><span class="pre">data</span></code>:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;site_label&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;D1&#39;</span><span class="p">,</span> <span class="s1">&#39;D1&#39;</span><span class="p">,</span> <span class="s1">&#39;A2&#39;</span><span class="p">,</span> <span class="s1">&#39;A2&#39;</span><span class="p">,</span> <span class="s1">&#39;F5&#39;</span><span class="p">,</span> <span class="s1">&#39;F5&#39;</span><span class="p">]</span>
<span class="n">data</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>site</th>
      <th>letter</th>
      <th>height</th>
      <th>color</th>
      <th>site_label</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>A</td>
      <td>1.0</td>
      <td>red</td>
      <td>D1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>C</td>
      <td>0.7</td>
      <td>gray</td>
      <td>D1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>C</td>
      <td>0.1</td>
      <td>gray</td>
      <td>A2</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2</td>
      <td>D</td>
      <td>1.2</td>
      <td>gray</td>
      <td>A2</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>A</td>
      <td>0.4</td>
      <td>red</td>
      <td>F5</td>
    </tr>
    <tr>
      <th>5</th>
      <td>5</td>
      <td>K</td>
      <td>0.4</td>
      <td>gray</td>
      <td>F5</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Now use <code class="docutils literal notranslate"><span class="pre">xtick_col</span></code> to set the xticks:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                            <span class="n">color_col</span><span class="o">=</span><span class="s1">&#39;color&#39;</span><span class="p">,</span>
                            <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_21_0.png" src="_images/basic_example_21_0.png" />
</div>
</div>
</div>
<div class="section" id="Shading-some-sites">
<h3>Shading some sites<a class="headerlink" href="#Shading-some-sites" title="Permalink to this headline">¶</a></h3>
<p>Sometimes we may want to shade certain sites. To do this, we specify the shade color and transparency (alpha) for all sites we want to shade (must be the same for all letters at a site):</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;shade_color&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;yellow&#39;</span><span class="p">,</span> <span class="s1">&#39;yellow&#39;</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="s1">&#39;brown&#39;</span><span class="p">,</span> <span class="s1">&#39;brown&#39;</span><span class="p">]</span>
<span class="n">data</span><span class="p">[</span><span class="s1">&#39;shade_alpha&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="mf">0.15</span><span class="p">,</span> <span class="mf">0.15</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="mf">0.3</span><span class="p">,</span> <span class="mf">0.3</span><span class="p">]</span>
<span class="n">data</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>site</th>
      <th>letter</th>
      <th>height</th>
      <th>color</th>
      <th>site_label</th>
      <th>shade_color</th>
      <th>shade_alpha</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>A</td>
      <td>1.0</td>
      <td>red</td>
      <td>D1</td>
      <td>yellow</td>
      <td>0.15</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>C</td>
      <td>0.7</td>
      <td>gray</td>
      <td>D1</td>
      <td>yellow</td>
      <td>0.15</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>C</td>
      <td>0.1</td>
      <td>gray</td>
      <td>A2</td>
      <td>None</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2</td>
      <td>D</td>
      <td>1.2</td>
      <td>gray</td>
      <td>A2</td>
      <td>None</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>A</td>
      <td>0.4</td>
      <td>red</td>
      <td>F5</td>
      <td>brown</td>
      <td>0.30</td>
    </tr>
    <tr>
      <th>5</th>
      <td>5</td>
      <td>K</td>
      <td>0.4</td>
      <td>gray</td>
      <td>F5</td>
      <td>brown</td>
      <td>0.30</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Now use <code class="docutils literal notranslate"><span class="pre">shade_color_col</span></code> and <code class="docutils literal notranslate"><span class="pre">shade_alpha_col</span></code> to draw the logo with the shaded sites:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                            <span class="n">color_col</span><span class="o">=</span><span class="s1">&#39;color&#39;</span><span class="p">,</span>
                            <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">,</span>
                            <span class="n">shade_color_col</span><span class="o">=</span><span class="s1">&#39;shade_color&#39;</span><span class="p">,</span>
                            <span class="n">shade_alpha_col</span><span class="o">=</span><span class="s1">&#39;shade_alpha&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_25_0.png" src="_images/basic_example_25_0.png" />
</div>
</div>
</div>
<div class="section" id="Adjusting-size,-axis-labels,-axes">
<h3>Adjusting size, axis labels, axes<a class="headerlink" href="#Adjusting-size,-axis-labels,-axes" title="Permalink to this headline">¶</a></h3>
<p>We can do additional formatting by scaling the width (<code class="docutils literal notranslate"><span class="pre">widthscale</span></code>), the height (<code class="docutils literal notranslate"><span class="pre">heightscale</span></code>), the axis font (<code class="docutils literal notranslate"><span class="pre">axisfontscale</span></code>), the x-axis (<code class="docutils literal notranslate"><span class="pre">xlabel</span></code>) and y-axis (<code class="docutils literal notranslate"><span class="pre">ylabel</span></code>) labels, and removing the axes altogether (<code class="docutils literal notranslate"><span class="pre">hide_axis</span></code>).</p>
<p>First, we make a plot where we adjust the size, change the y-axis label, and get rid of the x-axis label:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                            <span class="n">color_col</span><span class="o">=</span><span class="s1">&#39;color&#39;</span><span class="p">,</span>
                            <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">,</span>
                            <span class="n">xlabel</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">,</span>
                            <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;immune selection&#39;</span><span class="p">,</span>
                            <span class="n">heightscale</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                            <span class="n">axisfontscale</span><span class="o">=</span><span class="mf">1.5</span>
                            <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_27_0.png" src="_images/basic_example_27_0.png" />
</div>
</div>
<p>Now we make a plot where we hide the axes and their labels altogether:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                            <span class="n">color_col</span><span class="o">=</span><span class="s1">&#39;color&#39;</span><span class="p">,</span>
                            <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">,</span>
                            <span class="n">hide_axis</span><span class="o">=</span><span class="kc">True</span>
                            <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_29_0.png" src="_images/basic_example_29_0.png" />
</div>
</div>
</div>
<div class="section" id="Multiple-logos-in-one-figure">
<h3>Multiple logos in one figure<a class="headerlink" href="#Multiple-logos-in-one-figure" title="Permalink to this headline">¶</a></h3>
<p>So far we have made individual plots on newly generate figures created by <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.logo.html#dmslogo.logo.draw_logo">dmslogo.logo.draw_logo</a>.</p>
<p>But we can also create a multi-axis figure, and then draw several logos onto that. The easiest way to do this is with the <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.facet.html#dmslogo.facet.facet_plot">dmslogo.facet.facet_plot</a> command described below. But we can also do it using <code class="docutils literal notranslate"><span class="pre">matplotlib</span></code> subplots as here:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># make figure with two subplots: two rows, one column</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.3</span><span class="p">)</span> <span class="c1"># add more vertical space for axis titles</span>
<span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">)</span>

<span class="c1"># draw top plot, no x-axis ticks or label, default coloring</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">no_ticks</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">),</span>
                      <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                      <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                      <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                      <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
                      <span class="n">xlabel</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">,</span>
                      <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">,</span>
                      <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;no_ticks&#39;</span><span class="p">,</span>
                      <span class="n">title</span><span class="o">=</span><span class="s1">&#39;colored by amino acid&#39;</span><span class="p">)</span>

<span class="c1"># draw bottom plot, color as specified in `data`</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="p">,</span>
                      <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                      <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                      <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                      <span class="n">color_col</span><span class="o">=</span><span class="s1">&#39;color&#39;</span><span class="p">,</span>
                      <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span>
                      <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">,</span>
                      <span class="n">title</span><span class="o">=</span><span class="s1">&#39;user-specified colors&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_31_0.png" src="_images/basic_example_31_0.png" />
</div>
</div>
</div>
</div>
<div class="section" id="Real-HIV-data-from-Dingens-et-al">
<h2>Real HIV data from Dingens et al<a class="headerlink" href="#Real-HIV-data-from-Dingens-et-al" title="Permalink to this headline">¶</a></h2>
<p>In <a class="reference external" href="https://doi.org/10.1016/j.immuni.2018.12.017">An Antigenic Atlas of HIV-1 Escape from Broadly Neutralizing Antibodies Distinguishes Functional and Structural Epitopes</a> (Dingens et al, 2019), there are plots of immune selection on HIV envelope (Env) from anti-HIV antibodies at just a subset of “strongly selected” sites for each antibody.</p>
<p>Here we use <code class="docutils literal notranslate"><span class="pre">dmslogo</span></code> to re-create one of those plots (the one in Figure 3D,E) showing antibodies <em>PG9</em> and <em>PGT145</em>.</p>
<div class="section" id="Get-data-to-plot">
<h3>Get data to plot<a class="headerlink" href="#Get-data-to-plot" title="Permalink to this headline">¶</a></h3>
<p>We have already downloaded CSV files with the data from the paper’s <a class="reference external" href="https://github.com/jbloomlab/EnvsAntigenicAtlas/">GitHub repo</a> giving the immune selection (as <a class="reference external" href="https://jbloomlab.github.io/dms_tools2/fracsurvive.html">fraction surviving above average</a>) for these two antibodies We read the data into a pandas data frame:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">antibodies</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;PG9&#39;</span><span class="p">,</span> <span class="s1">&#39;PGT145&#39;</span><span class="p">]</span>

<span class="n">data_hiv</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">antibody</span> <span class="ow">in</span> <span class="n">antibodies</span><span class="p">:</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;input_files/summary_</span><span class="si">{</span><span class="n">antibody</span><span class="si">}</span><span class="s2">-medianmutfracsurvive.csv&quot;</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Reading data for </span><span class="si">{</span><span class="n">antibody</span><span class="si">}</span><span class="s2"> from </span><span class="si">{</span><span class="n">fname</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">data_hiv</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">antibody</span><span class="o">=</span><span class="n">antibody</span><span class="p">))</span>

<span class="n">data_hiv</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">data_hiv</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Reading data for PG9 from input_files/summary_PG9-medianmutfracsurvive.csv
Reading data for PGT145 from input_files/summary_PGT145-medianmutfracsurvive.csv
</pre></div></div>
</div>
<p>Here are the first few lines of the data frame. For each mutation it gives the immune selection (<em>mutfracsurvive</em>):</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data_hiv</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>site</th>
      <th>wildtype</th>
      <th>mutation</th>
      <th>mutfracsurvive</th>
      <th>antibody</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>160</td>
      <td>N</td>
      <td>I</td>
      <td>0.256342</td>
      <td>PG9</td>
    </tr>
    <tr>
      <th>1</th>
      <td>160</td>
      <td>N</td>
      <td>L</td>
      <td>0.207440</td>
      <td>PG9</td>
    </tr>
    <tr>
      <th>2</th>
      <td>160</td>
      <td>N</td>
      <td>R</td>
      <td>0.184067</td>
      <td>PG9</td>
    </tr>
    <tr>
      <th>3</th>
      <td>171</td>
      <td>K</td>
      <td>E</td>
      <td>0.176118</td>
      <td>PG9</td>
    </tr>
    <tr>
      <th>4</th>
      <td>428</td>
      <td>Q</td>
      <td>Y</td>
      <td>0.150981</td>
      <td>PG9</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>The sites in this data frame are in the <a class="reference external" href="https://www.hiv.lanl.gov/content/sequence/HIV/REVIEWS/HXB2.html">HXB2 numbering scheme</a>, which is <strong>not</strong> the same as sequential integer numbering of the actual BG505 Env for which the immune selection was measured. So for our plotting, we also need to create a column (which we will call <code class="docutils literal notranslate"><span class="pre">isite</span></code>) that numbers the sites a sequential numbering. A file that converts between HXB2 and and BG505 numbering is part of the paper’s <a class="reference external" href="https://github.com/jbloomlab/EnvsAntigenicAtlas/">GitHub
repo</a>; here we use a downloaded copy of that file to make the <code class="docutils literal notranslate"><span class="pre">isite</span></code> column:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">numberfile</span> <span class="o">=</span> <span class="s1">&#39;input_files/BG505_to_HXB2.csv&#39;</span>

<span class="n">data_hiv</span> <span class="o">=</span> <span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">numberfile</span><span class="p">)</span>
            <span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;original&#39;</span><span class="p">:</span><span class="s1">&#39;isite&#39;</span><span class="p">,</span> <span class="s1">&#39;new&#39;</span><span class="p">:</span><span class="s1">&#39;site&#39;</span><span class="p">})</span>
            <span class="p">[[</span><span class="s1">&#39;site&#39;</span><span class="p">,</span> <span class="s1">&#39;isite&#39;</span><span class="p">]]</span>
            <span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">data_hiv</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span> <span class="n">validate</span><span class="o">=</span><span class="s1">&#39;one_to_many&#39;</span><span class="p">)</span>
            <span class="p">)</span>
</pre></div>
</div>
</div>
<p>Now see how this data frame also has the <code class="docutils literal notranslate"><span class="pre">isite</span></code> column which has sequential integer numbering of the sequence:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data_hiv</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>site</th>
      <th>isite</th>
      <th>wildtype</th>
      <th>mutation</th>
      <th>mutfracsurvive</th>
      <th>antibody</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>Y</td>
      <td>0.030824</td>
      <td>PG9</td>
    </tr>
    <tr>
      <th>1</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>K</td>
      <td>0.006860</td>
      <td>PG9</td>
    </tr>
    <tr>
      <th>2</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>D</td>
      <td>0.006774</td>
      <td>PG9</td>
    </tr>
    <tr>
      <th>3</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>S</td>
      <td>0.004407</td>
      <td>PG9</td>
    </tr>
    <tr>
      <th>4</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>R</td>
      <td>0.003501</td>
      <td>PG9</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>We add a column (<code class="docutils literal notranslate"><span class="pre">site_label</span></code>) that gives the site labeled with the wildtype identity that we can use for axis ticks. We also indicate which sites to show (column <code class="docutils literal notranslate"><span class="pre">show_site</span></code>) in our logoplot snippet (these are just the same ones in Figure 3 of the <a class="reference external" href="https://doi.org/10.1016/j.immuni.2018.12.017">Dingens et al (2019)</a> paper):</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># same sites in Figure 3D,E of Dingens et al (2019)</span>
<span class="n">sites_to_show</span> <span class="o">=</span> <span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">119</span><span class="p">,</span> <span class="mi">125</span><span class="p">))</span> <span class="o">+</span>
                         <span class="p">[</span><span class="mi">127</span><span class="p">]</span> <span class="o">+</span>
                         <span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">156</span><span class="p">,</span> <span class="mi">174</span><span class="p">))</span> <span class="o">+</span>
                         <span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">199</span><span class="p">,</span> <span class="mi">205</span><span class="p">))</span> <span class="o">+</span>
                         <span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">312</span><span class="p">,</span> <span class="mi">316</span><span class="p">))</span>
                         <span class="p">)</span>

<span class="n">data_hiv</span> <span class="o">=</span> <span class="p">(</span>
    <span class="n">data_hiv</span>
    <span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">site_label</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="p">[</span><span class="s1">&#39;wildtype&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">x</span><span class="p">[</span><span class="s1">&#39;site&#39;</span><span class="p">],</span>
            <span class="n">show_site</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="p">[</span><span class="s1">&#39;site&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">sites_to_show</span><span class="p">),</span>
            <span class="p">)</span>
    <span class="p">)</span>
</pre></div>
</div>
</div>
<p>See how the data frame now has the <code class="docutils literal notranslate"><span class="pre">site_label</span></code> and <code class="docutils literal notranslate"><span class="pre">show_site</span></code> columns:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data_hiv</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>site</th>
      <th>isite</th>
      <th>wildtype</th>
      <th>mutation</th>
      <th>mutfracsurvive</th>
      <th>antibody</th>
      <th>site_label</th>
      <th>show_site</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>Y</td>
      <td>0.030824</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
    </tr>
    <tr>
      <th>1</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>K</td>
      <td>0.006860</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>D</td>
      <td>0.006774</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
    </tr>
    <tr>
      <th>3</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>S</td>
      <td>0.004407</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
    </tr>
    <tr>
      <th>4</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>R</td>
      <td>0.003501</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
</div>
<div class="section" id="Draw-a-logo-plot">
<h3>Draw a logo plot<a class="headerlink" href="#Draw-a-logo-plot" title="Permalink to this headline">¶</a></h3>
<p>Now we make logo plots of the sites that we have selected to show, here just for the <em>PG9</em> antibody. Note how we query <code class="docutils literal notranslate"><span class="pre">data_hiv</span></code> for only the antibody (<em>PG9</em>) and the sites of interest (<code class="docutils literal notranslate"><span class="pre">show_site</span></code> is <code class="docutils literal notranslate"><span class="pre">True</span></code>):</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data_hiv</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;antibody == &quot;PG9&quot;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;show_site&#39;</span><span class="p">),</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;isite&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;mutation&#39;</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;mutfracsurvive&#39;</span><span class="p">,</span>
                            <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">,</span>
                            <span class="n">title</span><span class="o">=</span><span class="s1">&#39;PG9&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_45_0.png" src="_images/basic_example_45_0.png" />
</div>
</div>
</div>
<div class="section" id="Draw-site-level-line-plots">
<h3>Draw site-level line plots<a class="headerlink" href="#Draw-site-level-line-plots" title="Permalink to this headline">¶</a></h3>
<p>The logo plot above shows selection at a subset of sites. But we might also want to summarize the selection across all sites (as is done in Figure 2 of <a class="reference external" href="https://doi.org/10.1016/j.immuni.2018.12.017">Dingens et al (2019)</a>).</p>
<p>An easy way to do this is to create a summary statistic at each site. Here we compute the total <a class="reference external" href="https://jbloomlab.github.io/dms_tools2/dms_tools2.fracsurvive.html">fracsurvive</a> at each site across all mutations, and add that to our data frame:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data_hiv</span> <span class="o">=</span> <span class="p">(</span>
    <span class="n">data_hiv</span>
    <span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;mutation != wildtype&#39;</span><span class="p">)</span> <span class="c1"># only care about mutations; get rid of wildtype values</span>
    <span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">totfracsurvive</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s1">&#39;antibody&#39;</span><span class="p">,</span> <span class="s1">&#39;site&#39;</span><span class="p">])</span>
                                        <span class="p">[</span><span class="s1">&#39;mutfracsurvive&#39;</span><span class="p">]</span>
                                        <span class="o">.</span><span class="n">transform</span><span class="p">(</span><span class="s1">&#39;sum&#39;</span><span class="p">)</span>
            <span class="p">)</span>
    <span class="p">)</span>
</pre></div>
</div>
</div>
<p>Now the data frame has a column (<code class="docutils literal notranslate"><span class="pre">totfracsurvive</span></code>) giving the average fraction surviving at each site:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data_hiv</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>site</th>
      <th>isite</th>
      <th>wildtype</th>
      <th>mutation</th>
      <th>mutfracsurvive</th>
      <th>antibody</th>
      <th>site_label</th>
      <th>show_site</th>
      <th>totfracsurvive</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>Y</td>
      <td>0.030824</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
      <td>0.062502</td>
    </tr>
    <tr>
      <th>1</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>K</td>
      <td>0.006860</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
      <td>0.062502</td>
    </tr>
    <tr>
      <th>2</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>D</td>
      <td>0.006774</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
      <td>0.062502</td>
    </tr>
    <tr>
      <th>3</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>S</td>
      <td>0.004407</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
      <td>0.062502</td>
    </tr>
    <tr>
      <th>4</th>
      <td>31</td>
      <td>30</td>
      <td>A</td>
      <td>R</td>
      <td>0.003501</td>
      <td>PG9</td>
      <td>A31</td>
      <td>False</td>
      <td>0.062502</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Now we use the <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.line.html#dmslogo.line.draw_line">dmslogo.line.draw_line</a> function to draw the line plot for antibody <em>PG9</em>. Note how we provide our new <code class="docutils literal notranslate"><span class="pre">totfracsurvive</span></code> column as <code class="docutils literal notranslate"><span class="pre">height_col</span></code>. We also provide our previously defined <code class="docutils literal notranslate"><span class="pre">show_site</span></code> column (which indicates which sites were shown in the logo plot) as the <code class="docutils literal notranslate"><span class="pre">show_col</span></code>, so that the line plot has the sites shown in the above logo plot underlined in orange:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_line</span><span class="p">(</span><span class="n">data_hiv</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;antibody == &quot;PG9&quot;&#39;</span><span class="p">),</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;isite&#39;</span><span class="p">,</span>
                            <span class="n">height_col</span><span class="o">=</span><span class="s1">&#39;totfracsurvive&#39;</span><span class="p">,</span>
                            <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                            <span class="n">show_col</span><span class="o">=</span><span class="s1">&#39;show_site&#39;</span><span class="p">,</span>
                            <span class="n">title</span><span class="o">=</span><span class="s1">&#39;PG9&#39;</span><span class="p">,</span>
                            <span class="n">widthscale</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_51_0.png" src="_images/basic_example_51_0.png" />
</div>
</div>
</div>
<div class="section" id="Combining-site-level-line-and-mutation-level-logo-plots">
<h3>Combining site-level line and mutation-level logo plots<a class="headerlink" href="#Combining-site-level-line-and-mutation-level-logo-plots" title="Permalink to this headline">¶</a></h3>
<p>Of course, a line plot isn’t that hard to make, but the advantage of doing this using the approach above is that we can combine <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.line.html#dmslogo.line.draw_line">dmslogo.line.draw_line</a> and <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.logo.html#dmslogo.logo.draw_logo">dmslogo.logo.draw_logo</a> to create a single figure that shows the site-selection in a line plot and the selected sites as logo plots.</p>
<p>The easiest way to do this using the <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.facet.html#dmslogo.facet.facet_plot">dmslogo.facet.facet_plot</a> command described below. But first here we do it using <code class="docutils literal notranslate"><span class="pre">matplotlib</span></code> subplots. Note how the resulting plot combines the line and logo plots, with the line plot using the orange underline to indicate which sites are zoomed in the logo plot:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span>
                         <span class="n">gridspec_kw</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;width_ratios&#39;</span><span class="p">:[</span><span class="mi">1</span><span class="p">,</span> <span class="mf">1.5</span><span class="p">]})</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.12</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">set_size_inches</span><span class="p">(</span><span class="mi">24</span><span class="p">,</span> <span class="mi">3</span><span class="p">)</span>

<span class="n">_</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_line</span><span class="p">(</span><span class="n">data_hiv</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;antibody == &quot;PG9&quot;&#39;</span><span class="p">),</span>
                      <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;isite&#39;</span><span class="p">,</span>
                      <span class="n">height_col</span><span class="o">=</span><span class="s1">&#39;totfracsurvive&#39;</span><span class="p">,</span>
                      <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                      <span class="n">show_col</span><span class="o">=</span><span class="s1">&#39;show_site&#39;</span><span class="p">,</span>
                      <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>

<span class="n">_</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data_hiv</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;antibody == &quot;PG9&quot;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;show_site&#39;</span><span class="p">),</span>
                      <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;isite&#39;</span><span class="p">,</span>
                      <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;mutation&#39;</span><span class="p">,</span>
                      <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;mutfracsurvive&#39;</span><span class="p">,</span>
                      <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span>
                      <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_53_0.png" src="_images/basic_example_53_0.png" />
</div>
</div>
</div>
</div>
<div class="section" id="Faceting-line-and-logo-plots-together">
<h2>Faceting line and logo plots together<a class="headerlink" href="#Faceting-line-and-logo-plots-together" title="Permalink to this headline">¶</a></h2>
<p>The easiest way to facet line and logo plots together is using <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.facet.html#dmslogo.facet.facet_plot">dmslogo.facet.facet_plot</a>.</p>
<p>The cell below shows how this is done. You pass the data to this function, as well any columns and rows we would like to facet, the <code class="docutils literal notranslate"><span class="pre">x_col</span></code> and <code class="docutils literal notranslate"><span class="pre">show_col</span></code> arguments shared between the line and logo plots, and additional keyword arguments for <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.logo.html#dmslogo.logo.draw_logo">dmslogo.logo.draw_logo</a> and <a class="reference external" href="https://jbloomlab.github.io/dmslogo/dmslogo.line.html#dmslogo.line.draw_line">dmslogo.line.draw_line</a>:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">facet_plot</span><span class="p">(</span>
                    <span class="n">data_hiv</span><span class="p">,</span>
                    <span class="n">gridrow_col</span><span class="o">=</span><span class="s1">&#39;antibody&#39;</span><span class="p">,</span>
                    <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;isite&#39;</span><span class="p">,</span>
                    <span class="n">show_col</span><span class="o">=</span><span class="s1">&#39;show_site&#39;</span><span class="p">,</span>
                    <span class="n">draw_line_kwargs</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">height_col</span><span class="o">=</span><span class="s1">&#39;totfracsurvive&#39;</span><span class="p">,</span>
                                          <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">),</span>
                    <span class="n">draw_logo_kwargs</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;mutation&#39;</span><span class="p">,</span>
                                          <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;mutfracsurvive&#39;</span><span class="p">,</span>
                                          <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">,</span>
                                          <span class="n">xlabel</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">),</span>
                    <span class="n">line_titlesuffix</span><span class="o">=</span><span class="s1">&#39;site-level selection&#39;</span><span class="p">,</span>
                    <span class="n">logo_titlesuffix</span><span class="o">=</span><span class="s1">&#39;mutation-level selection&#39;</span>
                    <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_55_0.png" src="_images/basic_example_55_0.png" />
</div>
</div>
<p>There are various options to tweak the formatting of the faceted plot. Here we demonstrate a few of them:</p>
<ul class="simple">
<li><p>We assign a more generic ylabel (“immune selection”) to each plot via the appropriate <code class="docutils literal notranslate"><span class="pre">*_kwargs</span></code> option.</p></li>
<li><p>We use the <code class="docutils literal notranslate"><span class="pre">share_ylim_across_rows=False</span></code> option to allow each row to have its own y-axis limits.</p></li>
<li><p>We use the <code class="docutils literal notranslate"><span class="pre">share_xlabel</span></code> and <code class="docutils literal notranslate"><span class="pre">share_ylabel</span></code> options to share the x- and y-labels across the line and logo plots.</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">facet_plot</span><span class="p">(</span>
                    <span class="n">data_hiv</span><span class="p">,</span>
                    <span class="n">gridrow_col</span><span class="o">=</span><span class="s1">&#39;antibody&#39;</span><span class="p">,</span>
                    <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;isite&#39;</span><span class="p">,</span>
                    <span class="n">show_col</span><span class="o">=</span><span class="s1">&#39;show_site&#39;</span><span class="p">,</span>
                    <span class="n">draw_line_kwargs</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">height_col</span><span class="o">=</span><span class="s1">&#39;totfracsurvive&#39;</span><span class="p">,</span>
                                          <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                                          <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;immune selection&#39;</span><span class="p">),</span>
                    <span class="n">draw_logo_kwargs</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;mutation&#39;</span><span class="p">,</span>
                                          <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;mutfracsurvive&#39;</span><span class="p">,</span>
                                          <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">,</span>
                                          <span class="n">xlabel</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                                          <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;immune selection&#39;</span><span class="p">),</span>
                    <span class="n">line_titlesuffix</span><span class="o">=</span><span class="s1">&#39;site-level selection&#39;</span><span class="p">,</span>
                    <span class="n">logo_titlesuffix</span><span class="o">=</span><span class="s1">&#39;mutation-level selection&#39;</span><span class="p">,</span>
                    <span class="n">share_ylim_across_rows</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                    <span class="n">share_xlabel</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                    <span class="n">share_ylabel</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                    <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_57_0.png" src="_images/basic_example_57_0.png" />
</div>
</div>
<p>Finally, here is the same plot where we have shaded the N-linked glycan at 160:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[30]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data_hiv</span><span class="p">[</span><span class="s1">&#39;shade_color&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">numpy</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">data_hiv</span><span class="p">[</span><span class="s1">&#39;site&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;160&#39;</span><span class="p">,</span> <span class="s1">&#39;161&#39;</span><span class="p">,</span> <span class="s1">&#39;162&#39;</span><span class="p">]),</span> <span class="s1">&#39;gray&#39;</span><span class="p">,</span> <span class="kc">None</span><span class="p">)</span>
<span class="n">data_hiv</span><span class="p">[</span><span class="s1">&#39;shade_alpha&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">numpy</span><span class="o">.</span><span class="n">where</span><span class="p">(</span><span class="n">data_hiv</span><span class="p">[</span><span class="s1">&#39;site&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;160&#39;</span><span class="p">,</span> <span class="s1">&#39;161&#39;</span><span class="p">,</span> <span class="s1">&#39;162&#39;</span><span class="p">]),</span> <span class="mf">0.15</span><span class="p">,</span> <span class="kc">None</span><span class="p">)</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">facet_plot</span><span class="p">(</span>
                    <span class="n">data_hiv</span><span class="p">,</span>
                    <span class="n">gridrow_col</span><span class="o">=</span><span class="s1">&#39;antibody&#39;</span><span class="p">,</span>
                    <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;isite&#39;</span><span class="p">,</span>
                    <span class="n">show_col</span><span class="o">=</span><span class="s1">&#39;show_site&#39;</span><span class="p">,</span>
                    <span class="n">draw_line_kwargs</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">height_col</span><span class="o">=</span><span class="s1">&#39;totfracsurvive&#39;</span><span class="p">,</span>
                                          <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                                          <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;immune selection&#39;</span><span class="p">),</span>
                    <span class="n">draw_logo_kwargs</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;mutation&#39;</span><span class="p">,</span>
                                          <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;mutfracsurvive&#39;</span><span class="p">,</span>
                                          <span class="n">xtick_col</span><span class="o">=</span><span class="s1">&#39;site_label&#39;</span><span class="p">,</span>
                                          <span class="n">xlabel</span><span class="o">=</span><span class="s1">&#39;site&#39;</span><span class="p">,</span>
                                          <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;immune selection&#39;</span><span class="p">,</span>
                                          <span class="n">shade_color_col</span><span class="o">=</span><span class="s1">&#39;shade_color&#39;</span><span class="p">,</span>
                                          <span class="n">shade_alpha_col</span><span class="o">=</span><span class="s1">&#39;shade_alpha&#39;</span><span class="p">),</span>
                    <span class="n">line_titlesuffix</span><span class="o">=</span><span class="s1">&#39;site-level selection&#39;</span><span class="p">,</span>
                    <span class="n">logo_titlesuffix</span><span class="o">=</span><span class="s1">&#39;mutation-level selection&#39;</span><span class="p">,</span>
                    <span class="n">share_ylim_across_rows</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                    <span class="n">share_xlabel</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                    <span class="n">share_ylabel</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                    <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_59_0.png" src="_images/basic_example_59_0.png" />
</div>
</div>
</div>
<div class="section" id="Some-details-about-fonts">
<h2>Some details about fonts<a class="headerlink" href="#Some-details-about-fonts" title="Permalink to this headline">¶</a></h2>
<div class="section" id="Write-DMSLOGO-in-Comic-Sans-font">
<h3>Write <em>DMSLOGO</em> in Comic Sans font<a class="headerlink" href="#Write-DMSLOGO-in-Comic-Sans-font" title="Permalink to this headline">¶</a></h3>
<p>Generate data to plot by creating the pandas DataFrame <code class="docutils literal notranslate"><span class="pre">word_data</span></code>. In this data frame, we choose large heights and bright colors for the letters in our word (<em>DMSLOGO</em>), and smaller letters and gray for other letters.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">word</span> <span class="o">=</span> <span class="s1">&#39;DMSLOGO&#39;</span>
<span class="n">lettercolors</span> <span class="o">=</span> <span class="p">[</span><span class="n">CBPALETTE</span><span class="p">[</span><span class="mi">1</span><span class="p">]]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="s1">&#39;dms&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="p">[</span><span class="n">CBPALETTE</span><span class="p">[</span><span class="mi">2</span><span class="p">]]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="s1">&#39;logo&#39;</span><span class="p">)</span>

<span class="c1"># make data frame with data to plot</span>
<span class="n">random</span><span class="o">.</span><span class="n">seed</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>
<span class="n">word_data</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;x&#39;</span><span class="p">:[],</span> <span class="s1">&#39;letter&#39;</span><span class="p">:[],</span> <span class="s1">&#39;height&#39;</span><span class="p">:[],</span> <span class="s1">&#39;color&#39;</span><span class="p">:[]}</span>
<span class="k">for</span> <span class="n">x</span><span class="p">,</span> <span class="p">(</span><span class="n">letter</span><span class="p">,</span> <span class="n">color</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">word</span><span class="p">,</span> <span class="n">lettercolors</span><span class="p">)):</span>
    <span class="n">word_data</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
    <span class="n">word_data</span><span class="p">[</span><span class="s1">&#39;letter&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">letter</span><span class="p">)</span>
    <span class="n">word_data</span><span class="p">[</span><span class="s1">&#39;color&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">color</span><span class="p">)</span>
    <span class="n">word_data</span><span class="p">[</span><span class="s1">&#39;height&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">random</span><span class="o">.</span><span class="n">uniform</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">1.5</span><span class="p">))</span>
    <span class="k">for</span> <span class="n">otherletter</span> <span class="ow">in</span> <span class="n">random</span><span class="o">.</span><span class="n">sample</span><span class="p">(</span><span class="nb">sorted</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="s1">&#39;ACTG&#39;</span><span class="p">)</span> <span class="o">-</span> <span class="p">{</span><span class="n">letter</span><span class="p">}),</span> <span class="mi">3</span><span class="p">):</span>
        <span class="n">word_data</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
        <span class="n">word_data</span><span class="p">[</span><span class="s1">&#39;letter&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">otherletter</span><span class="p">)</span>
        <span class="n">word_data</span><span class="p">[</span><span class="s1">&#39;color&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">CBPALETTE</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
        <span class="n">word_data</span><span class="p">[</span><span class="s1">&#39;height&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">random</span><span class="o">.</span><span class="n">uniform</span><span class="p">(</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">))</span>
<span class="n">word_data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">word_data</span><span class="p">)</span>
<span class="n">word_data</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="mi">6</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>x</th>
      <th>letter</th>
      <th>height</th>
      <th>color</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>D</td>
      <td>1.422211</td>
      <td>#E69F00</td>
    </tr>
    <tr>
      <th>1</th>
      <td>0</td>
      <td>T</td>
      <td>0.486186</td>
      <td>#999999</td>
    </tr>
    <tr>
      <th>2</th>
      <td>0</td>
      <td>A</td>
      <td>0.294371</td>
      <td>#999999</td>
    </tr>
    <tr>
      <th>3</th>
      <td>0</td>
      <td>C</td>
      <td>0.467294</td>
      <td>#999999</td>
    </tr>
    <tr>
      <th>4</th>
      <td>1</td>
      <td>M</td>
      <td>1.414926</td>
      <td>#E69F00</td>
    </tr>
    <tr>
      <th>5</th>
      <td>1</td>
      <td>T</td>
      <td>0.301875</td>
      <td>#999999</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Now draw the logo. We use the <code class="docutils literal notranslate"><span class="pre">fontfamily</span></code> argument to set a Comic Sans font. This also requires us to increase <code class="docutils literal notranslate"><span class="pre">fontaspect</span></code> since this font is wider, and increase <code class="docutils literal notranslate"><span class="pre">letterpad</span></code> as the font height sometimes sticks out beyond its baseline:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[32]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># NBVAL_IGNORE_OUTPUT</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dmslogo</span><span class="o">.</span><span class="n">draw_logo</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">word_data</span><span class="p">,</span>
                            <span class="n">letter_height_col</span><span class="o">=</span><span class="s1">&#39;height&#39;</span><span class="p">,</span>
                            <span class="n">x_col</span><span class="o">=</span><span class="s1">&#39;x&#39;</span><span class="p">,</span>
                            <span class="n">letter_col</span><span class="o">=</span><span class="s1">&#39;letter&#39;</span><span class="p">,</span>
                            <span class="n">color_col</span><span class="o">=</span><span class="s1">&#39;color&#39;</span><span class="p">,</span>
                            <span class="n">fontfamily</span><span class="o">=</span><span class="s1">&#39;Comic Sans MS&#39;</span><span class="p">,</span>
                            <span class="n">hide_axis</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                            <span class="n">fontaspect</span><span class="o">=</span><span class="mf">0.85</span><span class="p">,</span>
                            <span class="n">letterpad</span><span class="o">=</span><span class="mf">0.05</span>
                            <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/basic_example_63_0.png" src="_images/basic_example_63_0.png" />
</div>
</div>
</div>
<div class="section" id="Subtleties-of-non-default-fonts">
<h3>Subtleties of non-default fonts<a class="headerlink" href="#Subtleties-of-non-default-fonts" title="Permalink to this headline">¶</a></h3>
<p>Note however that you in general may have difficulty using most fonts (other than the <code class="docutils literal notranslate"><span class="pre">dmslogo</span></code> default) for good-looking logos. The reason is that for a clean and accurate letter-height logo plots, the font must:</p>
<ul class="simple">
<li><p>be mono-spaced</p></li>
<li><p>not have descenders</p></li>
<li><p>have all letters go exactly from the baseline to the top</p></li>
</ul>
<p>You can manually edit a font to do this as has been done for the current <code class="docutils literal notranslate"><span class="pre">dmslogo</span></code> default font; to see more information on this look <a class="reference external" href="https://github.com/jbloomlab/dmslogo/tree/master/dmslogo/ttf_fonts">here for details</a>.</p>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<p class="logo">
  <a href="index.html">
    <img class="logo" src="_static/BloomLogo.jpg" alt="Logo"/>
    
    <h1 class="logo logo-name">dmslogo</h1>
    
  </a>
</p>



<p class="blurb">Sequence logos for deep mutational scanning</p>




<p>
<iframe src="https://ghbtns.com/github-btn.html?user=jbloomlab&repo=dmslogo&type=watch&count=true&size=large&v=2"
  allowtransparency="true" frameborder="0" scrolling="0" width="200px" height="35px"></iframe>
</p>





    

<p>
<a class="badge" href="https://travis-ci.org/jbloomlab/dmslogo">
    <img
        alt="https://secure.travis-ci.org/jbloomlab/dmslogo.svg?branch=master"
        src="https://secure.travis-ci.org/jbloomlab/dmslogo.svg?branch=master"
    />
</a>
</p>


<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="examples.html">Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="dmslogo.html">dmslogo package</a></li>
<li class="toctree-l1"><a class="reference internal" href="package_index.html">package index</a></li>
<li class="toctree-l1"><a class="reference internal" href="acknowledgments.html">Acknowledgements</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="examples.html">Examples</a><ul>
      <li>Previous: <a href="examples.html" title="previous chapter">Examples</a></li>
      <li>Next: <a href="overlays.html" title="next chapter">Plotting heatmap overlays</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019--2021, Bloom lab.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 3.2.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/basic_example.nblink.txt"
          rel="nofollow">Page source</a>
    </div>

    
    <a href="https://github.com/jbloomlab/dmslogo" class="github">
        <img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png" alt="Fork me on GitHub"  class="github"/>
    </a>
    

    
  </body>
</html>